dht/i_partitioner.cc

```diff
@@ -179,6 +179,20 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_not_empty(const dht::token_range& r) {
+    bool is_bounded = r.start() && !r.start()->value().is_minimum() && r.end() && !r.end()->value().is_maximum();
+    if (is_bounded) {
+        uint64_t start_raw = r.start()->value().raw();
+        uint64_t end_raw = r.end()->value().raw();
+        return end_raw - start_raw >= (!r.start()->is_inclusive() + !r.end()->is_inclusive());
```

Comment:
You were careful to avoid overflow in two separate ways. Let's celebrate it with a comment, this isn't trivial.